iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/gcp/GcpPrivilegePolicyCheckPart.java

```diff
@@ -0,0 +1,123 @@
+import org.sonar.iac.terraform.api.tree.AttributeTree;
+import org.sonar.iac.terraform.api.tree.BlockTree;
+import org.sonar.iac.terraform.api.tree.FileTree;
+import org.sonar.iac.terraform.api.tree.TerraformTree;
+import org.sonar.iac.terraform.checks.AbstractNewResourceCheck;
+import org.sonar.iac.terraform.checks.utils.TerraformUtils;
+import org.sonar.iac.terraform.symbols.ResourceSymbol;
+
+import static org.sonar.iac.terraform.checks.AbstractResourceCheck.getResourceType;
+import static org.sonar.iac.terraform.checks.AbstractResourceCheck.isResource;
+
+public class GcpPrivilegePolicyCheckPart extends AbstractNewResourceCheck {
+
+  private static final String POLICY_MESSAGE = "Make sure it is safe to give all members full access.";
+  private static final String MEMBER_MESSAGE = "Make sure it is safe to grant that member full access.";
+  private static final String SECONDARY_MESSAGE = "The policy is used here.";
+
+  private final Map<String, AttributeTree> policyReferences = new HashMap<>();
+
+  @Override
+  public void initialize(InitContext init) {
+    super.initialize(init);
+    init.register(FileTree.class, (ctx, tree) -> new PolicyReferenceCollector().scan(new TreeContext(), tree));
+  }
+
+  @Override
+  protected void provideResource(CheckContext ctx, BlockTree blockTree) {
+    super.provideResource(ctx, blockTree);
+    if (isData(blockTree)) {
+      ResourceSymbol data = ResourceSymbol.fromPresent(ctx, blockTree);
+      if ("google_iam_policy".equals(data.type)) {
+        checkPolicy(data);
+      }
+    }
+  }
+
+  private static boolean isData(BlockTree blockTree) {
+    return "data".equals(blockTree.key().value());
+  }
+
+  private void checkPolicy(ResourceSymbol data) {
+    AttributeTree reference = policyReferences.get(formatReference(data));
+    if (reference != null) {
+      SecondaryLocation secondary = new SecondaryLocation(reference, SECONDARY_MESSAGE);
+      data.blocks("binding").forEach(
+        binding -> binding.attribute("role")
+          .reportIf(matchesPattern(".*(?:admin|manager|owner|superuser).*"), POLICY_MESSAGE, secondary));
+    }
+  }
+
+  private static String formatReference(ResourceSymbol policy) {
+    return String.format("data.google_iam_policy.%s.policy_data", policy.name);
+  }
+
+  @Override
+  protected void registerResourceConsumer() {
+    register(List.of("google_project_iam_binding", "google_organization_iam_binding",
+        "google_service_account_iam_binding", "google_folder_iam_binding"),
+      resource -> checkRole(resource, POLICY_MESSAGE));
+
+    register(List.of("google_project_iam_member", "google_organization_iam_member",
+        "google_service_account_iam_member", "google_folder_iam_member"),
+      resource -> checkRole(resource, MEMBER_MESSAGE));
+  }
+
+  private void checkRole(ResourceSymbol resource, String message) {
+    resource.attribute("role")
+      .reportIf(matchesPattern(".*(?:admin|manager|owner|superuser).*"), message);
```

Comment:
Same suggestion here:
".*\b(?:admin|manager|owner|superuser)\b.*"
(Otherwise managerial and administrative will also be matched.)

Reply:
See above